common/unsafe/src/test/java/org/apache/spark/unsafe/types/CollationSupportSuite.java

```diff
@@ -101,6 +101,9 @@ public void testContains() throws SparkException {
     assertContains("ab世De", "AB世dE", "UNICODE_CI", true);
     assertContains("äbćδe", "ÄbćδE", "UNICODE_CI", true);
     assertContains("äbćδe", "ÄBcΔÉ", "UNICODE_CI", false);
+    // Case-variable character length
```

Comment:
I would say more precisely "Longer binary lowercase representation", or something like that, since in future we may add test cases for "Longer binary uppercase representation" and such comments would point out the difference for a reader.

Reply:
Actually, do you maybe wanna add a test case for `'ß'.upper() == 'SS'`? Since it's a CI, and not LOWERCASE collation.

Reply:
[ICU StringSearch](https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/text/StringSearch.html) UNICODE_CI doesn't recognize this

as per these docs, I'd expect this to work only with a "GERMAN_CI" (which we don't yet have in Spark)
```
StringSearch ensures that language eccentricity can be handled, e.g. for the German collator, characters ß and SS will be matched if case is chosen to be ignored. See the ["ICU Collation Design Document"](https://htmlpreview.github.io/?https://github.com/unicode-org/icu-docs/blob/main/design/collation/ICU_collation_design.htm) for more information.
```